pandas/core/internals/blocks.py

```diff
@@ -2341,30 +2341,26 @@ def get_block_type(dtype: DtypeObj):
     -------
     cls : class, subclass of Block
     """
-    # We use kind checks because it is much more performant
-    #  than is_foo_dtype
-    kind = dtype.kind
-
-    cls: type[Block]
-
     if isinstance(dtype, SparseDtype):
         # Need this first(ish) so that Sparse[datetime] is sparse
-        cls = ExtensionBlock
+        return ExtensionBlock
     elif isinstance(dtype, DatetimeTZDtype):
-        cls = DatetimeTZBlock
+        return DatetimeTZBlock
     elif isinstance(dtype, PeriodDtype):
-        cls = NDArrayBackedExtensionBlock
+        return NDArrayBackedExtensionBlock
     elif isinstance(dtype, ExtensionDtype):
         # Note: need to be sure PandasArray is unwrapped before we get here
-        cls = ExtensionBlock
+        return ExtensionBlock
 
-    elif kind in ["M", "m"]:
-        cls = DatetimeLikeBlock
+    # We use kind checks because it is much more performant
+    #  than is_foo_dtype
+    kind = dtype.kind
+    if kind in ["M", "m"]:
+        return DatetimeLikeBlock
     elif kind in ["f", "c", "i", "u", "b"]:
```

Comment:
we can improve a little bit here by checking `kind in "fciub"` instead of the list

Reply:
updated